plugins/modules/jenkins_plugin.py

```diff
@@ -430,11 +443,21 @@ def _get_url_data(
         if msg_exception is None:
             msg_exception = "Retrieval of %s failed." % what
 
+        # Get any custom headers from kwargs, default to empty dictionary if none
+        custom_headers = kwargs.get('headers', {})
+
+        # Merge custom headers with self.crumb, ensuring self.crumb is always included
+        headers = {**self.crumb, **custom_headers}
+
+        # If force_basic_auth is True, add the Authorization header
+        if self.force_basic_auth:
+            headers["Authorization"] = self._create_basic_auth_header()
```

Comment:
I don't think this assignment/computation is necessary since `fetch_url` is already doing that.